iterations-Katharina601/functions/data_transformations.R

```diff
@@ -0,0 +1,89 @@
+transform_metadata_to_df <-function(stations_metadata)
+
+ {
+  # Convert the latestData to UTC format
+  stations_metadata$latestData <- as.POSIXct(stations_metadata$latestData, tz = "UTC")
+  
+  # Create a data frame
+  stations_metadata_df <- data.frame(
+    id = stations_metadata$id,
+    name = stations_metadata$name,
+    latestData = stations_metadata$latestData,
+    lat = stations_metadata$lat,
+    lon = stations_metadata$lon
+  )
+  
+  return(stations_metadata_df)
+}
+
+stations_metadata[[1]] %>%  
+  map_dfr(as_tibble)  %>% 
+  mutate(latestData = map_chr(latestData, 1, .default = NA_character_)) %>% 
+  mutate(latestData = as_datetime(latestData, tz = "UTC"))  %>% 
+  unnest_wider(location) %>% 
+  unnest_wider(latLon)

```

Comment:
This code works, but you the code inside the function transform_metadata_to_df() doesnt work. Making a function from the code you have written here would be correct